<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Tony Chen" />


<title>Microbiome Project</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Cystic-Fibrosis-and-Gut-Microbiome</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/jiezhou-2/Cystic-Fibrosis-and-Gut-Microbiome">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Microbiome Project</h1>
<h4 class="author">Tony Chen</h4>
<h4 class="date">11/9/2021</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-10
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 2
</p>
<p>
<strong>Knit directory:</strong> <code>Cystic-Fibrosis-and-Gut-Microbiome/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is untracked by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181210code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181210code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181210)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
C:/Users/Jie Zhou/Documents/paper02052022/longitudinalOutcome/Cystic-Fibrosis-and-Gut-Microbiome/data/ddata.csv
</td>
<td style="text-align:left;">
data/ddata.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
C:/Users/Jie Zhou/Documents/paper02052022/longitudinalOutcome/Cystic-Fibrosis-and-Gut-Microbiome/data/analyst_pulmonary_exacerbation_202110061618.csv
</td>
<td style="text-align:left;">
data/analyst_pulmonary_exacerbation_202110061618.csv
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomjiezhou2CysticFibrosisandGutMicrobiometree4eefd1fcc04fce56973f2daa54eca62dc0c2bb06targetblank4eefd1fa"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/jiezhou-2/Cystic-Fibrosis-and-Gut-Microbiome/tree/4eefd1fcc04fce56973f2daa54eca62dc0c2bb06" target="_blank">4eefd1f</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomjiezhou2CysticFibrosisandGutMicrobiometree4eefd1fcc04fce56973f2daa54eca62dc0c2bb06targetblank4eefd1fa" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/jiezhou-2/Cystic-Fibrosis-and-Gut-Microbiome/tree/4eefd1fcc04fce56973f2daa54eca62dc0c2bb06" target="_blank">4eefd1f</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .RData
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  analysis/marginal model.Rmd
    Untracked:  analysis/pca of relative abundance of microbiome.Rmd
    Untracked:  analysis/pca-of-relative-abundance-of-microbiome.Rmd
    Untracked:  analysis/random effect model.Rmd
    Untracked:  analysis/survival model.Rmd

Unstaged changes:
    Modified:   analysis/index.Rmd
    Deleted:    analysis/longitudinal_analysis.Rmd
    Deleted:    analysis/model.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="input-the-microbiome-data-38-subjects-82-microbes.-these-data-have-undergone-log-ratio-transfomation-and-so-can-be-deemed-as-normal-data.-as-for-the-outcome-data-also-is-input-in-a-separate-file." class="section level3">
<h3>Input the microbiome data: 38 subjects, 82 microbes. These data have undergone log-ratio transfomation and so can be deemed as normal data. As for the Outcome data also is input in a separate file.</h3>
<pre class="r"><code>taxa_raw = read.csv(&quot;C:/Users/Jie Zhou/Documents/paper02052022/longitudinalOutcome/Cystic-Fibrosis-and-Gut-Microbiome/data/ddata.csv&quot;)
pulm = read.delim(&quot;C:/Users/Jie Zhou/Documents/paper02052022/longitudinalOutcome/Cystic-Fibrosis-and-Gut-Microbiome/data/analyst_pulmonary_exacerbation_202110061618.csv&quot;)
variable.names(pulm)</code></pre>
<pre><code> [1] &quot;patient_id&quot;                          &quot;event_type&quot;                         
 [3] &quot;event_date&quot;                          &quot;id&quot;                                 
 [5] &quot;action_on&quot;                           &quot;action_by&quot;                          
 [7] &quot;inserted_on&quot;                         &quot;inserted_by&quot;                        
 [9] &quot;record_comment&quot;                      &quot;enrollment_id&quot;                      
[11] &quot;followup_visit_id&quot;                   &quot;pe_comment&quot;                         
[13] &quot;pe_occurance&quot;                        &quot;pe_symptom&quot;                         
[15] &quot;oracle_itemnum&quot;                      &quot;user_delete&quot;                        
[17] &quot;pe_date&quot;                             &quot;pe_treatment&quot;                       
[19] &quot;pe_hospitalized&quot;                     &quot;pe_sx_wheeze_with_upper_resp_inf&quot;   
[21] &quot;pe_sx_wheeze_during_night_early_am&quot;  &quot;pe_sx_wheeze_all_the_time&quot;          
[23] &quot;pe_sx_hemoptysis&quot;                    &quot;pe_sx_pneumothorax&quot;                 
[25] &quot;pe_sx_chr_sinus_inf&quot;                 &quot;pe_sx_other&quot;                        
[27] &quot;pe_sx_specified&quot;                     &quot;pe_sx_incr_nas_cong_drain&quot;          
[29] &quot;pe_sx_fatigue&quot;                       &quot;pe_sx_decr_appetite&quot;                
[31] &quot;pe_sx_decr_oxy&quot;                      &quot;pe_sx_fever&quot;                        
[33] &quot;pe_sx_incr_br&quot;                       &quot;pe_fev1&quot;                            
[35] &quot;pe_tx_inhaled_antibiotics&quot;           &quot;pe_tx_iv_antibiotics&quot;               
[37] &quot;pe_tx_oral_antibiotics&quot;              &quot;pe_tx_chest_pt&quot;                     
[39] &quot;pe_tx_mucolytics&quot;                    &quot;pe_tx_dnase&quot;                        
[41] &quot;pe_tx_inhaled_saline&quot;                &quot;pe_tx_anti_inflammatory_medications&quot;
[43] &quot;pe_tx_steroids&quot;                      &quot;pe_tx_optimizing_nutrition&quot;         
[45] &quot;pe_tx_glucose_control&quot;               &quot;pe_tx_exercise&quot;                     
[47] &quot;pe_tx_supplemental_oxygen&quot;           &quot;pe_tx_ventilation&quot;                  
[49] &quot;pe_tx_ecmo&quot;                          &quot;pe_tx_antivirals&quot;                   
[51] &quot;pe_tx_analgesics&quot;                    &quot;pe_tx_antipyretics&quot;                 
[53] &quot;pe_tx_albuterol&quot;                     &quot;pe_tx_other&quot;                        
[55] &quot;pe_tx_specified&quot;                     &quot;pe_sx_incr_cough&quot;                   </code></pre>
<pre class="r"><code># averaging across time points
taxa_avg = taxa_raw %&gt;% 
  group_by(subject) %&gt;% 
  summarize_all(list(mean))

taxa = taxa_avg[,4:ncol(taxa_avg)]</code></pre>
</div>
<div id="selecting-the-microbial-interaction-network-using-glasso.-here-the-tuning-parameter-is-recommended-based-on-previous-study.-given-the-selected-network-leadeigen-algorithm-is-employed-to-find-the-communities-in-the-network." class="section level3">
<h3>Selecting the microbial interaction network using glasso. Here the tuning parameter is recommended based on previous study. Given the selected network, leadeigen algorithm is employed to find the communities in the network.</h3>
<pre class="r"><code>set.seed(2)
# lambda_glasso = seq(0.05, 6, 0.01)
# cv_glasso_taxa = CVglasso(taxa, K = 5, lam = lambda_glasso)
#   lam_pick = cv_glasso_taxa$Tuning[2]
lam_pick=0.75
glasso_taxa = glasso(cov(taxa), rho = lam_pick)
pcov_taxa = glasso_taxa$wi
taxa_nwk = as.matrix(ifelse(pcov_taxa != 0, 1, 0))
# graph theory

taxa_graph = graph.adjacency(taxa_nwk, mode = &quot;undirected&quot;)
taxa_louvain = cluster_louvain(taxa_graph)
taxa_leadeigen = cluster_leading_eigen(taxa_graph) # This one produces fewer modules (clusters)</code></pre>
</div>
<div id="for-each-of-the-communities-compute-its-principal-components." class="section level3">
<h3>For each of the communities, compute its principal components.</h3>
<pre class="r"><code>colnames(taxa[which(taxa_leadeigen$membership == 3)])</code></pre>
<pre><code> [1] &quot;Erysipelatoclostridium&quot; &quot;Bacteroides&quot;            &quot;Parabacteroides&quot;       
 [4] &quot;Tyzzerella&quot;             &quot;Mediterraneibacter&quot;     &quot;Sellimonas&quot;            
 [7] &quot;Flavonifractor&quot;         &quot;Ruminococcus2&quot;          &quot;Ruthenibacterium&quot;      
[10] &quot;Alistipes&quot;              &quot;Clostridium_XVIII&quot;      &quot;Monoglobus&quot;            
[13] &quot;Eggerthella&quot;            &quot;Turicibacter&quot;           &quot;Eisenbergiella&quot;        
[16] &quot;Flintibacter&quot;           &quot;Agathobaculum&quot;          &quot;Anaerotignum&quot;          
[19] &quot;Dysosmobacter&quot;          &quot;Bilophila&quot;              &quot;Clostridium_IV&quot;        
[22] &quot;Faecalicatena&quot;         </code></pre>
<pre class="r"><code>taxa_le_cluster1 = taxa[which(taxa_leadeigen$membership == 1)]
taxa_le_pca1 = prcomp(taxa_le_cluster1)
taxa_le_eigen1 = as.matrix(taxa_le_cluster1) %*% taxa_le_pca1$rotation
plot(taxa_le_pca1$sdev^2/sum(taxa_le_pca1$sdev^2), type=&quot;l&quot;, main = &quot;Leading Eigen Cluster 1 PCs&quot;, 
     xlab=&quot;principle components&quot;, ylab=&quot;variance explained&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-1.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>plot(taxa_le_eigen1[,1], taxa_le_eigen1[,2], main = &quot;First cluster&quot;, 
     xlab=&quot;PCA1&quot;, ylab=&quot;PCA2&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-2.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>summary(taxa_le_pca1)</code></pre>
<pre><code>Importance of components:
                          PC1     PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     9.7474 3.04649 2.85256 2.46476 2.28825 2.14601 2.02753
Proportion of Variance 0.6113 0.05971 0.05235 0.03908 0.03369 0.02963 0.02645
Cumulative Proportion  0.6113 0.67099 0.72334 0.76242 0.79611 0.82574 0.85219
                           PC8     PC9    PC10    PC11    PC12    PC13    PC14
Standard deviation     1.83713 1.69675 1.49252 1.43282 1.39194 1.30321 1.20289
Proportion of Variance 0.02171 0.01852 0.01433 0.01321 0.01247 0.01093 0.00931
Cumulative Proportion  0.87390 0.89242 0.90675 0.91996 0.93243 0.94335 0.95266
                          PC15    PC16    PC17    PC18    PC19    PC20    PC21
Standard deviation     1.08809 0.95990 0.91115 0.87864 0.79410 0.75140 0.66717
Proportion of Variance 0.00762 0.00593 0.00534 0.00497 0.00406 0.00363 0.00286
Cumulative Proportion  0.96028 0.96621 0.97155 0.97652 0.98057 0.98421 0.98707
                          PC22    PC23    PC24    PC25    PC26    PC27    PC28
Standard deviation     0.63713 0.59364 0.55947 0.53479 0.44205 0.39067 0.30968
Proportion of Variance 0.00261 0.00227 0.00201 0.00184 0.00126 0.00098 0.00062
Cumulative Proportion  0.98968 0.99195 0.99396 0.99580 0.99706 0.99804 0.99866
                          PC29    PC30    PC31    PC32    PC33
Standard deviation     0.29779 0.23523 0.19061 0.15304 0.06929
Proportion of Variance 0.00057 0.00036 0.00023 0.00015 0.00003
Cumulative Proportion  0.99923 0.99958 0.99982 0.99997 1.00000</code></pre>
<pre class="r"><code>taxa_le_cluster2 = taxa[which(taxa_leadeigen$membership == 2)]
taxa_le_pca2 = prcomp(taxa_le_cluster2)
taxa_le_eigen2 = as.matrix(taxa_le_cluster2) %*% taxa_le_pca2$rotation
plot(taxa_le_pca2$sdev^2/sum(taxa_le_pca2$sdev^2), type=&quot;l&quot;, main = &quot;Leading Eigen Cluster 2 PCs&quot;, 
     xlab=&quot;principle components&quot;, ylab=&quot;variance explained&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-3.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>plot(taxa_le_eigen2[,1], taxa_le_eigen2[,2], main = &quot;Second cluster&quot;, 
     xlab=&quot;PCA1&quot;, ylab=&quot;PCA2&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-4.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>summary(taxa_le_pca2)</code></pre>
<pre><code>Importance of components:
                          PC1    PC2     PC3     PC4     PC5     PC6     PC7
Standard deviation     9.0411 4.8374 4.08913 3.63476 3.44137 3.09360 2.82121
Proportion of Variance 0.3961 0.1134 0.08102 0.06401 0.05738 0.04637 0.03857
Cumulative Proportion  0.3961 0.5094 0.59047 0.65448 0.71187 0.75824 0.79680
                           PC8     PC9    PC10    PC11    PC12    PC13    PC14
Standard deviation     2.71755 2.40777 2.07910 2.03113 1.85601 1.77656 1.63739
Proportion of Variance 0.03578 0.02809 0.02094 0.01999 0.01669 0.01529 0.01299
Cumulative Proportion  0.83259 0.86068 0.88162 0.90161 0.91830 0.93360 0.94659
                          PC15    PC16    PC17    PC18    PC19    PC20    PC21
Standard deviation     1.44161 1.24244 1.22847 1.15619 1.08041 0.89318 0.80377
Proportion of Variance 0.01007 0.00748 0.00731 0.00648 0.00566 0.00387 0.00313
Cumulative Proportion  0.95666 0.96414 0.97145 0.97793 0.98358 0.98745 0.99058
                          PC22   PC23    PC24    PC25    PC26    PC27
Standard deviation     0.74307 0.6581 0.60150 0.51462 0.44220 0.37021
Proportion of Variance 0.00268 0.0021 0.00175 0.00128 0.00095 0.00066
Cumulative Proportion  0.99325 0.9953 0.99711 0.99839 0.99934 1.00000</code></pre>
<pre class="r"><code>taxa_le_cluster3= taxa[which(taxa_leadeigen$membership == 3)]
taxa_le_pca3 = prcomp(taxa_le_cluster3)
taxa_le_eigen3 = as.matrix(taxa_le_cluster3) %*% taxa_le_pca3$rotation
plot(taxa_le_pca3$sdev^2/sum(taxa_le_pca3$sdev^2), type=&quot;l&quot;, main = &quot;Leading Eigen Cluster 3 PCs&quot;, 
     xlab=&quot;principle components&quot;, ylab=&quot;variance explained&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-5.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>plot(taxa_le_eigen3[,1], taxa_le_eigen3[,2], main = &quot;Third cluster&quot;, 
     xlab=&quot;PCA1&quot;, ylab=&quot;PCA2&quot;)</code></pre>
<img src="figure/pca%20of%20relative%20abundance%20of%20microbiome.Rmd/unnamed-chunk-3-6.png" width="672" style="display: block; margin: auto;" />
<div class="alert alert-warning">
<strong>Warning!</strong> The custom <code>fig.path</code> you set was ignored by workflowr.
</div>
<pre class="r"><code>summary(taxa_le_pca3)</code></pre>
<pre><code>Importance of components:
                          PC1    PC2     PC3    PC4    PC5     PC6     PC7
Standard deviation     7.3329 3.3965 3.20515 2.8028 2.5060 2.04305 1.94407
Proportion of Variance 0.4743 0.1018 0.09062 0.0693 0.0554 0.03682 0.03334
Cumulative Proportion  0.4743 0.5761 0.66674 0.7360 0.7914 0.82826 0.86160
                           PC8     PC9    PC10    PC11    PC12   PC13    PC14
Standard deviation     1.76180 1.64773 1.49647 1.41514 1.11306 1.0542 0.92463
Proportion of Variance 0.02738 0.02395 0.01976 0.01767 0.01093 0.0098 0.00754
Cumulative Proportion  0.88898 0.91294 0.93269 0.95036 0.96129 0.9711 0.97863
                          PC15    PC16    PC17    PC18    PC19    PC20   PC21
Standard deviation     0.88317 0.68642 0.59050 0.50156 0.44078 0.39739 0.3842
Proportion of Variance 0.00688 0.00416 0.00308 0.00222 0.00171 0.00139 0.0013
Cumulative Proportion  0.98551 0.98967 0.99274 0.99496 0.99668 0.99807 0.9994
                          PC22
Standard deviation     0.26658
Proportion of Variance 0.00063
Cumulative Proportion  1.00000</code></pre>
</div>
<div id="linking-outcome" class="section level1">
<h1>Linking Outcome</h1>
<p>These are the pulmonary exacerbation events that have been entered into the database. As you will notice, there are often multiple events per person.</p>
<p>Not all patients or patient IDs on this list have a 16S microbiome sequence.</p>
<p>Note that pe_occurence=1 - is a definite yes; 2 is no (but was previously entered as having a PE due to some mild PE symptoms) and 3 is unknown. As a sensitivity analysis, you can exclude 2 and 3 to see if your deductions change.</p>
<p>If pe_occurence is missing, you can exclude unless the comments indicate otherwise.</p>
<p>Dates are sometimes missing when the pe_date was unknown; use event_date instead,</p>
<pre class="r"><code>taxa_short = taxa_avg %&gt;% 
  filter(subject %in% unique(pulm$patient_id))
taxa_short$subject</code></pre>
<pre><code> [1] 103 104 109 110 111 113 114 116 118 119 120 122 124 125 126 129 131 132</code></pre>
<pre class="r"><code>pulm_new = pulm %&gt;% group_by(patient_id) %&gt;% 
  filter(patient_id %in% (taxa_avg$subject[unique(taxa_avg$subject) %in%
                                          unique(pulm$patient_id)]))
pulm_new$pe_occurance = ifelse(pulm_new$pe_occurance == 1, 1, 0)
pulm_new = pulm_new %&gt;% 
  summarise(pulmonary = sum(pe_occurance, na.rm = T))
colnames(pulm_new) = c(&quot;subject&quot;, &quot;pe&quot;)

taxa_merge = left_join(taxa_short, pulm_new, by = &quot;subject&quot;)

#View(taxa_merge[c(&quot;subject&quot;, &quot;pe&quot;)])</code></pre>
</div>
<div id="poisson-regression-first-pcs-for-each-of-three-clusters" class="section level1">
<h1>Poisson regression: first pcs for each of three clusters</h1>
<pre class="r"><code># Get eigentaxa
index_final = which(taxa_avg$subject %in% taxa_merge$subject)
## first pc for each cluster
taxa_pois = as.data.frame(cbind(taxa_le_eigen1[index_final,1], 
      taxa_le_eigen2[index_final,1],
      taxa_le_eigen3[index_final,1],
      taxa_merge$pe,
      ifelse(taxa_merge$pe == 0, 0, 1)))
colnames(taxa_pois) = c(&quot;taxa_eigen1&quot;, &quot;taxa_eigen2&quot;, &quot;taxa_eigen3&quot;,
                        &quot;pe&quot;, &quot;pe_bi&quot;)
summary(glm(pe ~ taxa_eigen1 + taxa_eigen2 + taxa_eigen3, family = &quot;poisson&quot;, control = glm.control(maxit = 50),data = taxa_pois))</code></pre>
<pre><code>
Call:
glm(formula = pe ~ taxa_eigen1 + taxa_eigen2 + taxa_eigen3, family = &quot;poisson&quot;, 
    data = taxa_pois, control = glm.control(maxit = 50))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.5930  -0.6063  -0.2631   0.4297   1.4278  

Coefficients:
              Estimate Std. Error z value Pr(&gt;|z|)  
(Intercept)  0.7103003  0.9064313   0.784   0.4333  
taxa_eigen1  0.0682120  0.0292852   2.329   0.0198 *
taxa_eigen2 -0.0670223  0.0324842  -2.063   0.0391 *
taxa_eigen3  0.0007441  0.0525040   0.014   0.9887  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 18.995  on 17  degrees of freedom
Residual deviance: 11.391  on 14  degrees of freedom
AIC: 58.988

Number of Fisher Scoring iterations: 5</code></pre>
</div>
<div id="poisson-regression-first-pcs-for-each-of-three-clusters-1" class="section level1">
<h1>Poisson regression: first pcs for each of three clusters</h1>
<pre class="r"><code># Get eigentaxa
index_final = which(taxa_avg$subject %in% taxa_merge$subject)
## first pc for each cluster
taxa_pois = as.data.frame(cbind(taxa_le_eigen1[index_final,1], taxa_le_eigen1[index_final,2], 
      taxa_le_eigen2[index_final,1],
      taxa_le_eigen2[index_final,2], 
      taxa_le_eigen3[index_final,1],
      taxa_le_eigen3[index_final,2], 
      taxa_merge$pe,
      ifelse(taxa_merge$pe == 0, 0, 1)))
colnames(taxa_pois) = c(&quot;taxa_eigen11&quot;,&quot;taxa_eigen12&quot;, &quot;taxa_eigen21&quot;,&quot;taxa_eigen22&quot;, &quot;taxa_eigen31&quot;,&quot;taxa_eigen32&quot;,
                        &quot;pe&quot;, &quot;pe_bi&quot;)
summary(glm(pe ~ taxa_eigen11 + taxa_eigen12+ taxa_eigen21 +  taxa_eigen22+ taxa_eigen31+ taxa_eigen32, family = &quot;poisson&quot;, control = glm.control(maxit = 50),data = taxa_pois))</code></pre>
<pre><code>
Call:
glm(formula = pe ~ taxa_eigen11 + taxa_eigen12 + taxa_eigen21 + 
    taxa_eigen22 + taxa_eigen31 + taxa_eigen32, family = &quot;poisson&quot;, 
    data = taxa_pois, control = glm.control(maxit = 50))

Deviance Residuals: 
    Min       1Q   Median       3Q      Max  
-1.6485  -0.4101  -0.2211   0.3466   1.3566  

Coefficients:
             Estimate Std. Error z value Pr(&gt;|z|)  
(Intercept)   0.94751    1.38892   0.682   0.4951  
taxa_eigen11  0.05834    0.04325   1.349   0.1773  
taxa_eigen12 -0.04583    0.13470  -0.340   0.7337  
taxa_eigen21 -0.08593    0.04017  -2.139   0.0324 *
taxa_eigen22 -0.02758    0.08126  -0.339   0.7343  
taxa_eigen31 -0.03590    0.08427  -0.426   0.6701  
taxa_eigen32 -0.15042    0.15247  -0.987   0.3239  
---
Signif. codes:  0 &#39;***&#39; 0.001 &#39;**&#39; 0.01 &#39;*&#39; 0.05 &#39;.&#39; 0.1 &#39; &#39; 1

(Dispersion parameter for poisson family taken to be 1)

    Null deviance: 18.995  on 17  degrees of freedom
Residual deviance: 10.380  on 11  degrees of freedom
AIC: 63.977

Number of Fisher Scoring iterations: 5</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] parallel  stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] corrplot_0.92        glmnet_4.1-3         Matrix_1.3-4        
 [4] glasso_1.11          sna_2.6              network_1.17.1      
 [7] statnet.common_4.5.0 GGally_2.1.2         intergraph_2.0-2    
[10] ggplot2_3.3.5        igraph_1.2.11        CVglasso_1.0        
[13] doParallel_1.0.17    iterators_1.0.14     foreach_1.5.2       
[16] tidyr_1.2.0          magrittr_2.0.2       dplyr_1.0.8         
[19] data.table_1.14.2   

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8         lattice_0.20-45    rprojroot_2.0.2    digest_0.6.29     
 [5] utf8_1.2.2         R6_2.5.1           plyr_1.8.6         evaluate_0.14     
 [9] coda_0.19-4        highr_0.9          pillar_1.7.0       rlang_1.0.1       
[13] rstudioapi_0.13    jquerylib_0.1.4    rmarkdown_2.11     splines_4.1.2     
[17] stringr_1.4.0      munsell_0.5.0      compiler_4.1.2     httpuv_1.6.5      
[21] xfun_0.29          pkgconfig_2.0.3    shape_1.4.6        htmltools_0.5.2   
[25] tidyselect_1.1.1   tibble_3.1.6       workflowr_1.7.0    codetools_0.2-18  
[29] reshape_0.8.8      fansi_1.0.2        crayon_1.5.0       withr_2.4.3       
[33] later_1.3.0        grid_4.1.2         jsonlite_1.7.3     gtable_0.3.0      
[37] lifecycle_1.0.1    git2r_0.29.0       scales_1.1.1       cli_3.1.1         
[41] stringi_1.7.6      fs_1.5.2           promises_1.2.0.1   bslib_0.3.1       
[45] ellipsis_0.3.2     generics_0.1.2     vctrs_0.3.8        RColorBrewer_1.1-2
[49] tools_4.1.2        glue_1.6.1         purrr_0.3.4        survival_3.2-13   
[53] fastmap_1.1.0      yaml_2.2.2         colorspace_2.0-2   knitr_1.37        
[57] sass_0.4.0        </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
