<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Siming Zhao" />


<title>Power analysis based on simulations</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DiffDriver</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/szhaolab/diff_driver_analysis">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Power analysis based on simulations</h1>
<h4 class="author">Siming Zhao</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-04-03
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 5 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>diff_driver_analysis-main/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20181210code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20181210)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20181210code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20181210)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongnoversioncontrol"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>Repository version:</strong> no version control </a>
</p>
</div>
<div id="strongRepositoryversionstrongnoversioncontrol" class="panel-collapse collapse">
<div class="panel-body">
<p>Tracking code development and connecting the code version to the results is critical for reproducibility. To start using Git, open the Terminal and type <code>git init</code> in your project directory.</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
This project is not being versioned with Git. To obtain the full reproducibility benefits of using workflowr, please see <code>?wflow_start</code>.
</p>
<hr>
</div>
</div>
</div>
<div id="methods" class="section level2">
<h2>Methods</h2>
<p>There are several approaches to identify genes associated with certain phenotypes.</p>
<div id="multi-linear-regression-anova" class="section level3">
<h3>1. Multi-linear regression (ANOVA)</h3>
<p>This was the approach used by domainXplorer (<a href="http://cancerimmunolres.aacrjournals.org/content/early/2016/07/09/2326-6066.CIR-15-0233">Porta-Pardo and Godzik, 2016</a>). The software is available <a href="https://github.com/eduardporta/domainXplorer">here</a>.</p>
<p>Let <span class="math inline">\(E_i\)</span> be some form of immune score for tumor sample <span class="math inline">\(i\)</span>, <span class="math inline">\(T_i\)</span> is the tumor type for sample <span class="math inline">\(i\)</span>, <span class="math inline">\(N_i\)</span> is the total number of mutations in sample <span class="math inline">\(i\)</span> (or mutation rate)., <span class="math inline">\(D_{jg}\)</span> is the binary indicator of the mutation status for gene <span class="math inline">\(g\)</span> in sample <span class="math inline">\(i\)</span>. For each <span class="math inline">\(i\)</span>, we have the following model (for simplicity, omiting <span class="math inline">\(i\)</span>)</p>
<p><span class="math display">\[ E = \beta_0 + \beta_1 * T + \beta_2* N + \beta_3 *D_g. \]</span> In the next step, <em>p</em> value is obtained by ANOVA test.</p>
<p>In <a href="https://www.sciencedirect.com/science/article/pii/S0092867418303131?via%3Dihub">Ding et al 2018 cell</a>, domainXplorer was used to identify driver events that correlate with the presence of different immune cell types. So <span class="math inline">\(E\)</span> is the cell type fraction. In <a href="https://www.sciencedirect.com/science/article/pii/S0092867414016390?via%3Dihub">Rooney et al 2015 cell</a>, <span class="math inline">\(E\)</span> is rank transformed CYT (cytolytic activity defined based on gene expression).</p>
</div>
<div id="modeling-of-count-data-countlrt" class="section level3">
<h3>2. Modeling of count data (CountLRT)</h3>
<p>We propose a new approach to address this problem. Let <span class="math inline">\(Y_{ij}\)</span> be the number of mutations at position <span class="math inline">\(j\)</span>, sample <span class="math inline">\(i\)</span>, <span class="math inline">\(Y_{ij} \in \{0,1\}\)</span>. <span class="math inline">\(\mu_{ij}\)</span> is background mutation rate at position <span class="math inline">\(j\)</span>, sample <span class="math inline">\(i\)</span>.</p>
<p><span class="math display">\[Y_{ij} \sim Bernoulli(\mu_{ij}\delta_{ij})\]</span> Here <span class="math inline">\(\delta_{ij}\)</span> is the deviation of observed mutation rate from the background mutation rate. It is dependent on the sample’s immune score <span class="math inline">\(E_i\)</span> in the following way:</p>
<p><span class="math display">\[ log(\delta_{ij})=\beta_{ij}^0 + \beta_g*E_{i} \]</span></p>
<p><span class="math inline">\(\beta_g\)</span> is gene <span class="math inline">\(g\)</span>’s coefficient for <span class="math inline">\(E_i\)</span>. For the null model, <span class="math inline">\(\beta_g=0\)</span>. For the alternative model, we could get Maximum Likelihood Estimation for <span class="math inline">\(\beta_g\)</span> and get Bayes Factor for gene <span class="math inline">\(g\)</span>.</p>
<p>In our approach, we can take into account confounders by modeling <span class="math inline">\(\mu_{ij}\)</span> and <span class="math inline">\(\beta_{ij}^0\)</span>.</p>
</div>
<div id="binomial-test" class="section level3">
<h3>3. Binomial test</h3>
<p>This method only works for binary <span class="math inline">\(E_i\)</span> data. In this approach, we aggregate all positions in a gene and count the number of mutations in <span class="math inline">\(E_i=1\)</span> (<span class="math inline">\(N_1\)</span>) and <span class="math inline">\(E_i=0\)</span> (<span class="math inline">\(N_2\)</span>) groups, we assess if <span class="math inline">\(N_1\)</span> deviates from <span class="math inline">\(Binom(N_1+N_2, \frac{\#\{E_i=1\}} {\#\{E_i=1\} + \#\{E_i=0\}})\)</span>.</p>
</div>
<div id="logistic-regression-gene-level" class="section level3">
<h3>4. logistic regression (gene level)</h3>
<p>This method only works for binary <span class="math inline">\(E_i\)</span> data.</p>
<p><span class="math display">\[log(\frac{p(E_i=1)}{1-p(E_i=1)})= \beta * D_i +\beta_0\]</span></p>
</div>
<div id="fishers-exact-test" class="section level3">
<h3>5. Fisher’s exact test</h3>
<p>This method only works for binary <span class="math inline">\(E_i\)</span> data. In this approach, we aggregate all positions in a gene and count the number of mutations in <span class="math inline">\(E_i=1\)</span> (<span class="math inline">\(N_1\)</span>) and <span class="math inline">\(E_i=0\)</span> (<span class="math inline">\(N_2\)</span>) groups then use Fisher exact test a p value.</p>
</div>
</div>
<div id="simulation" class="section level2">
<h2>Simulation</h2>
<p>In this simulation, we assume no gene specific effect of background mutation rate, i.e. no <span class="math inline">\(\lambda_g\)</span> term for background mutation rate. Within the same gene, we assume all non-silent positions share the same selection mutation rate across positions, i.e does not distinguish functional positions from the rest non-silent positions.</p>
<p><strong>Generate data</strong></p>
<ul>
<li><p>Generate <span class="math inline">\(E_i\)</span>. For a given sample size <span class="math inline">\(N\)</span>, we assign <span class="math inline">\(N/2\)</span> samples with <span class="math inline">\(E=1\)</span>, the rest have <span class="math inline">\(E=0\)</span></p></li>
<li><p>Generate <span class="math inline">\(Y_{ij}\)</span>. We randomly pick one gene, <em>ERBB3</em>. For <span class="math inline">\(Y_{ij}\)</span> in this gene, we simulate silent mutations based on realistic parameters estimated from UCS data using driverMAPS. We simulate nonsilent mutations with <span class="math inline">\(\beta_{ij}^0=0\)</span> (this implies that on average this gene is not selected) or <span class="math inline">\(\beta_{ij}^0=1\)</span> (this implies that on average this gene is positively selected). We simulate with <span class="math inline">\(\beta_{g(j)}=\{0,0.2,0.8\}\)</span>. <span class="math inline">\(\beta_{g(j)}=0\)</span> indicates that this gene is not associated with the phenotype. Repeat this process 1000 times.</p></li>
</ul>
<p><strong>Run analysis</strong><br />
We run five methods for each of the 5000 times and obtain p values (for method 2, p values are obtained by likelihood ratio tests). Compare number of rounds with p value &lt;0.01 for these methods.</p>
<p>We assume all samples come from the same tumor type and the mutation rate in each sample is the same. For method 2, we didn’t estimate mutation rate for each sample, but used the mutation rate from simulation.</p>
</div>
<div id="results" class="section level2">
<h2>Results</h2>
<div id="gene-is-not-associated-with-phenotype-beta_g0-." class="section level3">
<h3>gene is not associated with phenotype ( <span class="math inline">\(\beta_g=0\)</span> ).</h3>
<p>In this simulation, the mutation data was generated with <span class="math inline">\(\beta_{ij}^0=0\)</span> and <span class="math inline">\(\beta_g=0\)</span>. We treat all position <span class="math inline">\(i\)</span> with the same relative risk,i.e. didn’t account for different functional impacts of among positions.</p>
<div class="figure">
<img src="../figures/power_betaf0=0_betagc=0.png" alt="" />
<p class="caption">null</p>
</div>
</div>
<div id="gene-with-low-mutation-rate-and-weakly-associated-with-phenotype-beta_ij00-and-beta_g0.2-." class="section level3">
<h3>gene with low mutation rate and weakly associated with phenotype ( <span class="math inline">\(\beta_{ij}^0=0\)</span> and <span class="math inline">\(\beta_g=0.2\)</span> ).</h3>
<p>In this simulation, the mutation data was generated with <span class="math inline">\(\beta_{ij}^0=0\)</span> and <span class="math inline">\(\beta_g=0.2\)</span>.</p>
<div class="figure">
<img src="../figures/power_betaf0=0_betagc=0.2.png" alt="" />
<p class="caption">lowweak</p>
</div>
</div>
<div id="gene-with-low-mutation-rate-and-strongly-associated-with-phenotype-beta_ij00-and-beta_g0.8-." class="section level3">
<h3>gene with low mutation rate and strongly associated with phenotype ( <span class="math inline">\(\beta_{ij}^0=0\)</span> and <span class="math inline">\(\beta_g=0.8\)</span> ).</h3>
<p>In this simulation, the mutation data was generated with <span class="math inline">\(\beta_{ij}^0=0\)</span> and <span class="math inline">\(\beta_g=0.8\)</span>.</p>
<div class="figure">
<img src="../figures/power_betaf0=0_betagc=0.8.png" alt="" />
<p class="caption">lowstrong</p>
</div>
</div>
<div id="gene-with-high-mutation-rate-and-weakly-associated-with-phenotype-beta_ij01-and-beta_g0.2-." class="section level3">
<h3>gene with high mutation rate and weakly associated with phenotype ( <span class="math inline">\(\beta_{ij}^0=1\)</span> and <span class="math inline">\(\beta_g=0.2\)</span> ).</h3>
<p>In this simulation, the mutation data was generated with <span class="math inline">\(\beta_{ij}^0=1\)</span> and <span class="math inline">\(\beta_g=0.2\)</span>.</p>
<div class="figure">
<img src="../figures/power_betaf0=1_betagc=0.2.png" alt="" />
<p class="caption">highweak</p>
</div>
</div>
<div id="gene-with-high-mutation-rate-and-strongly-associated-with-phenotype-beta_ij01-and-beta_g0.8-." class="section level3">
<h3>gene with high mutation rate and strongly associated with phenotype ( <span class="math inline">\(\beta_{ij}^0=1\)</span> and <span class="math inline">\(\beta_g=0.8\)</span> ).</h3>
<p>In this simulation, the mutation data was generated with <span class="math inline">\(\beta_{ij}^0=1\)</span> and <span class="math inline">\(\beta_g=0.8\)</span>.</p>
<div class="figure">
<img src="../figures/power_betaf0=1_betagc=0.8.png" alt="" />
<p class="caption">highstrong</p>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.2 (2021-11-01)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19043)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.8       rstudioapi_0.13  knitr_1.37       magrittr_2.0.2  
 [5] workflowr_1.7.0  R6_2.5.1         rlang_1.0.1      fastmap_1.1.0   
 [9] fansi_1.0.2      stringr_1.4.0    tools_4.1.2      xfun_0.29       
[13] utf8_1.2.2       cli_3.1.1        git2r_0.29.0     jquerylib_0.1.4 
[17] htmltools_0.5.2  ellipsis_0.3.2   rprojroot_2.0.2  yaml_2.2.2      
[21] digest_0.6.29    tibble_3.1.6     lifecycle_1.0.1  crayon_1.5.0    
[25] later_1.3.0      sass_0.4.0       vctrs_0.3.8      promises_1.2.0.1
[29] fs_1.5.2         glue_1.6.1       evaluate_0.14    rmarkdown_2.11  
[33] stringi_1.7.6    bslib_0.3.1      compiler_4.1.2   pillar_1.7.0    
[37] jsonlite_1.7.3   httpuv_1.6.5     pkgconfig_2.0.3 </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
